packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -129,6 +129,11 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
   Uri getTestUri(Uri key, int version) => key;
 
   /// Calculate the appropriate basedir for the current test context.
+  ///
+  /// The optional [suffix] argument is used by the
+  /// [FlutterSkiaGoldFileComparator] and the [FlutterPreSubmitFileComparator].
+  /// These [FlutterGoldenFileComparators] randomize their base directories to
+  /// maintain thread safety while using the `goldctl` tool.
```

Comment:
Why aren't we just using `createTempSync` from package:file?

https://pub.dev/documentation/file/latest/file/Directory/createTempSync.html

Reply:
When run locally, we want the files to be accessible to see diffs etc.

Reply:
Ahh.  I'm a bit curious about whether we'd hit collisions at some point using math.random, but it's probably fine.

Reply:
I thought so too, but then realized we only run 2-3 processes concurrently on ci, and only ~130/~4500 tests are golden file tests, so I feel like a collision would be incredibly rare in how this is written.